enclave/src/attestation.rs

```diff
@@ -434,6 +441,14 @@ pub fn create_attestation_report<A: EnclaveAttestationOCallApi>(
     // p_qe_report and report.data to confirm the QUOTE has not be modified and
     // is not a replay. It is optional.
 
+    // need to call this a second time (first time is when we get the sigrl revocation list)
+    // (has some internal state that needs to be reset)!
+    let ias_socket_result = ocall_api.ocall_get_ias_socket();
+    if ias_socket_result.0 != sgx_status_t::SGX_SUCCESS {
+        return Err(ias_socket_result.0);
+    }
+    let ias_socket = ias_socket_result.1;
+
```

Comment:
During the refactoring, this looked like a code duplication - but apparently it is necessary to call `get_ias_socket()` a second time. 
We should refactor this function, split it into multiple functions with appropriate names to make it less error prone.

Reply:
Oh interessing, we even discussed a potential internal state. :)